src/jit/flowgraph.cpp

```diff
@@ -23965,23 +23965,31 @@ void Compiler::fgMergeFinallyChains()
         return;
     }
 
+    bool enableMergeFinallyChains = true;
+
```

Comment:
The  optimizer might still issue the unreachable warning for this way to doing this.
Let me check